factcast-store-validation/src/test/java/org/factcast/store/pgsql/internal/listen/PgConnectionSupplierTest.java

```diff
@@ -0,0 +1,164 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+public class PgConnectionSupplierTest {
+
+    PgConnectionSupplier uut;
+
+    @Mock
+    private org.apache.tomcat.jdbc.pool.DataSource ds;
+
+    @BeforeEach
+    void setUp() {
+        initMocks(this);
+        uut = new PgConnectionSupplier(ds);
+    }
+
+    private void setupConnectionProperties(String properties) {
+        PoolConfiguration poolConf = mock(PoolConfiguration.class);
+        when(poolConf.getConnectionProperties()).thenReturn(properties);
+        when(ds.getPoolProperties()).thenReturn(poolConf);
+    }
+
+    @Test
+    void test_wrongDataSourceImplementation() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            DataSource ds = mock(DataSource.class);
+            new PgConnectionSupplier(ds);
+            failBecauseExceptionWasNotThrown(IllegalStateException.class);
+        });
+    }
+
+    @Test
+    void test_buildCredentials() {
+        // given
+        PoolConfiguration poolConf = mock(PoolConfiguration.class);
+        when(poolConf.getPassword()).thenReturn("testPassword");
+        when(poolConf.getUsername()).thenReturn("testUsername");
+        when(ds.getPoolProperties()).thenReturn(poolConf);
+        // when
+        Properties creds = uut.buildPgConnectionProperties(ds);
+        // then
+        assertEquals("testUsername", creds.get("user"));
+        assertEquals("testPassword", creds.get("password"));
+    }
+
+    @Test
+    void test_socketTimeout() {
+        // given
+        setupConnectionProperties("socketTimeout=30;connectTimeout=20;loginTimeout=10;");
+        // when
+        Properties connectionProperties = uut.buildPgConnectionProperties(ds);
+        // then
+        assertEquals("30", connectionProperties.get("socketTimeout"));
+    }
+
+    @Test
+    void test_connectionTimeout() {
+        // given
+        setupConnectionProperties("socketTimeout=30;connectTimeout=20;loginTimeout=10;");
+        // when
+        Properties connectionProperties = uut.buildPgConnectionProperties(ds);
+        // then
+        assertEquals("20", connectionProperties.get("connectTimeout"));
+    }
+
+    @Test
+    void test_loginTimeout() {
+        // given
+        setupConnectionProperties("socketTimeout=30;connectTimeout=20;loginTimeout=10;");
+        // when
+        Properties connectionProperties = uut.buildPgConnectionProperties(ds);
+        // then
+        assertEquals("10", connectionProperties.get("loginTimeout"));
+    }
+
+    @Test
+    void test_noRelevantPropertySet() {
+        setupConnectionProperties("sockettimeout=30");
+        uut.buildPgConnectionProperties(ds);
+    }
+
+    @Test
+    void test_propertySyntaxBroken() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            setupConnectionProperties("sockettimeout:30");
+            uut.buildPgConnectionProperties(ds);
+        });
+    }
+
+    @Test
+    void test_nullProperties() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'test_nullProperties' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)